core/src/main/scala/org/apache/spark/Partition.scala

```diff
@@ -26,6 +26,11 @@ trait Partition extends Serializable {
    */
   def index: Int
 
+  override def equals(other: Any): Boolean = other match {
+    case that: Partition => super.equals(that) && that.index == index
```

Comment:
If so, do you mean using the `canEqual` approach?
If not, do you mean removing both `equals` and `hashCode` then?